### DISCUSSION TITLE
fiona

### DISCUSSION THREAD

**Discussion by JCxOx** [flet-dev/flet#4417] [Packages] [answered]
Hello and thanks for flet dev !

I am developing an Android application to draw polygones and polylines from shapefile (SHP) on a map.
Flet.map is fine, and we can select many maps (openstreetmap, google, orthophotos ...)
That works well on my computer but not on Android device because the "fiona" module in not found.

https://pypi.org/project/fiona/

I am using the last version of flet 0.25.0 dev3750 & python 3.12.7

In case "fiona" cannot be used with flet (android), other packages to read shapefile are available :
geopandas , shapely, GDAL.

Thank you very much,
JC

**Comment by ndonkoHenri** [comment]
Try building your app using geopandas.

**Reply by JCxOx** [reply]
Thanks for your prompt reply,

I will try it, I will post the result asap

**Reply by JCxOx** [reply]
Some news:

I have changed my python code to process the Shapefile with geopandas only (no more fiona).
The problem is similar, as geopandas is not available for flet :

I use the following command to build the apk :
    $ flet build apk --include-packages flet_map geopandas
Error message :
Because map_ortho_shp depends on geopandas any which doesn't exist (could not find package geopandas at                 
           https://pub.dev), version solving failed. 

If you have any idea to solve the problem, thanks,

JC

**Reply by FeodorFitsner** [reply]
`geopandas` should go to requirements.txt (pyproject.toml for pre-release).

**Reply by JCxOx** [reply]
Hello,

Thanks for your quick answer, but I am not sure to understand correctly,
There is a pyproject.toml file in my project folder. What I have to do with this file ? It seems there is no reference to a requirements.txt file ...

If you have any tips or example please don't hesitate,

Thanks,

JC

**Reply by FeodorFitsner** [reply]
can you paste the contents of your `pyproject.toml` here?

**Reply by JCxOx** [reply]
The content of pyproject.toml is :

```
[project]
name = "map_ortho_shp"
version = "0.1.0"
description = ""
readme = "README.md"
requires-python = ">=3.8"
dependencies = [
  "flet==0.25.0.dev3750"
]


[tool.flet]
# org name in reverse domain name notation, e.g. "com.mycompany".
# Combined with project.name to build bundle ID for iOS and Android apps
org = "com.mycompany"

# project display name that is used as an app title on Android and iOS home screens,
# shown in window titles and about app dialogs on desktop.
product = "map_ortho_shp"

# company name to display in about app dialogs
company = "Flet"

# copyright text to display in about app dialogs
copyright = "Copyright (C) 2024 by Flet"

[tool.flet.app]
path = "src"
```

**Reply by JCxOx** [reply]
```
[project]
name = "map_ortho_shp"
version = "0.1.0"
description = ""
readme = "README.md"
requires-python = ">=3.8"
dependencies = [
  "flet==0.25.0.dev3750"
]


[tool.flet]
# org name in reverse domain name notation, e.g. "com.mycompany".
# Combined with project.name to build bundle ID for iOS and Android apps
org = "com.mycompany"

# project display name that is used as an app title on Android and iOS home screens,
# shown in window titles and about app dialogs on desktop.
product = "map_ortho_shp"

# company name to display in about app dialogs
company = "Flet"

# copyright text to display in about app dialogs
copyright = "Copyright (C) 2024 by Flet"

[tool.flet.app]
path = "src"
```

**Reply by JCxOx** [reply]
Sorry, I copied 2 times the contents, and with a poor format ...

**Reply by FeodorFitsner** [reply]
OK, add `geopandas` to `dependencies`:

```
dependencies = [
  "flet==0.25.0.dev3750",
  "geopandas"
]
```

and remove from `--include-packages`.

**Reply by JCxOx** [reply]
Ok thanks !

**Reply by JCxOx** [reply]
New pyproject.toml file :

[project]
name = "map_ortho_shp"
version = "0.1.0"
description = ""
readme = "README.md"
requires-python = ">=3.8"
dependencies = [
  "flet==0.25.0.dev3750",
  "geopandas"
]
...

Command :

$ flet build apk --include-packages flet_map

Results :

(base) jc@jc-HP-Compaq-8200-Elite-SFF-PC:~/Documents/Flet/map_ortho_shp$ flet build apk --include-packages flet_map
[22:42:54] Created Flutter bootstrap project from gh:flet-dev/flet-build-template with ref 0.25.0 ✅                                            
           Customized app icons and splash images ✅                                                                                            
[22:43:04] Generated app icons ✅                                                                                                               
[22:43:12] Generated splash screens ✅                                                                                                          
[22:43:37] Running package command                                                                                                              
           Extra PyPi indexes:                                                                                                                  
           Creating asset directory: /home/jc/Documents/Flet/map_ortho_shp/build/flutter/app                                                    
           Created temp directory: /tmp/serious_python_tempGGSSJC                                                                               
           Copying Python app from /home/jc/Documents/Flet/map_ortho_shp/src to a temp directory                                                
           Cleanup app                                                                                                                          
           Configured Android/arm64-v8a platform with sitecustomize.py                                                                          
           Installing  with pip command to /home/jc/Documents/Flet/map_ortho_shp/build/site-packages/arm64-v8a                                  
           Downloading Python distributive from                                                                                                 
           https://github.com/indygreg/python-build-standalone/releases/download/20240909/cpython-3.12.6+20240909-x86_64-unknown-linux-gnu-insta
           ll_only_stripped.tar.gz to a build directory                                                                                         
           Extracting Python distributive                                                                                                       
                                                                                                                                                
           ERROR: Cannot install geopandas==0.10.0, geopandas==0.10.1, geopandas==0.10.2, geopandas==0.11.0, geopandas==0.11.1,                 
           geopandas==0.12.0, geopandas==0.12.1, geopandas==0.12.2, geopandas==0.13.0, geopandas==0.13.1, geopandas==0.13.2, geopandas==0.14.0, 
           geopandas==0.14.1, geopandas==0.14.2, geopandas==0.14.3, geopandas==0.14.4, geopandas==0.2.1, geopandas==0.3.0, geopandas==0.4.0,    
           geopandas==0.4.1, geopandas==0.5.0, geopandas==0.5.1, geopandas==0.6.0, geopandas==0.6.1, geopandas==0.6.2, geopandas==0.6.3,        
           geopandas==0.7.0, geopandas==0.8.0, geopandas==0.8.1, geopandas==0.8.2, geopandas==0.9.0, geopandas==1.0.0 and geopandas==1.0.1      
           because these package versions have conflicting dependencies.                                                                        
           ERROR: ResolutionImpossible: for help visit                                                                                          
           https://pip.pypa.io/en/latest/topics/dependency-resolution/#dealing-with-dependency-conflicts                                        
                                                                                                                                                
           Error building Flet app - see the log of failed command above.                                                                       
[22:43:41] Doctor summary (to see all details, run flutter doctor -v):                                                                          
           [✓] Flutter (Channel stable, 3.24.5, on Linux Mint 20.1 5.4.0-144-generic, locale fr_FR.UTF-8)                                       
           [✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)                                                     
           [✓] Chrome - develop for the web                                                                                                     
           [✓] Linux toolchain - develop for Linux desktop                                                                                      
           [✓] Android Studio (version 2022.3)                                                                                                  
           [✓] VS Code (version 1.95.3)                                                                                                         
           [✓] Connected device (2 available)                                                                                                   
           [✓] Network resources                                                                                                                
                                                                                                                                                
           • No issues found!

**Reply by FeodorFitsner** [reply]
Alright, I see `geopandas` depends on at least `pyogrio`, `pyproj` and `shapely` binary packages. So, `geopandas` can't be used now. Will look into adding those packages later.

**Reply by JCxOx** [reply]
Ok, thank you very much for your help !

JC

**Reply by JCxOx** [reply]
Hello,

I have finally found a pure python module to read SHP file : 

https://pypi.org/project/pyshp/

Now I can build the APK without any errors, everything looks clean.
The application starts correctly and displays the map frame (only) and buttons ... but no map inside frame ! whatever the tile type (openstreetmap, google) ... nothing is displayed in the map frame (it is always gray).

The map tiles come from specific url (one for openstreetmap, one for Google ...) and I don't know if I need to change any permission to allow internet access ? if yes, how to do it ?

Thanks a lot,
JC

**Reply by ndonkoHenri** [reply]
Might be related to https://github.com/flet-dev/flet/issues/4362

**Reply by JCxOx** [reply]
I had a look at #4362, and that seems similar indeed, but there is no solution for the map issue on Android with Flet 0.25 ...

Thanks,
JC

**Reply by FeodorFitsner** [reply]
pyogrio, pyproj, shapely and geopandas are all available.

**Reply by FeodorFitsner** [reply, marked as answer]
fiona and GDAL are available too.

**Reply by JCxOx** [reply]
Hello,

Thank you very much ! good news.

**Comment by FeodorFitsner** [comment]
Flet notes: fiona package depends on GDAL: https://github.com/OSGeo/gdal

**Reply by JCxOx** [reply]
Thanks for your quick answer,

I will try with geopandas as proposed by ndonkoHenri.
